IfcMobileTelecommunicationsAppliance Rail Import BIMQ 2021-07-21T06:43

jmirtsch opened the pull request:
Rail properties for IfcMobileTelecommunicationsAppliance

jmirtsch committed:
IfcMobileTelecommunicationsAppliance Rail Import BIMQ 2021-07-21T06:43:55.7379174Z

czapplitec commented on IFC4x3/Properties/e/ExchangeCapacity_3BKPzAHt10s8UrVNVqgpxw/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="ExchangeCapacity_3BKPzAHt10s8UrVNVqgpxw" Name="ExchangeCapacity" UniqueId="cb519f4a-4770-40d8-87b5-7d77f4ab3efa" PrimaryDataType="IfcInteger" />
```
No need for change

czapplitec commented on IFC4x3/Properties/m/MaximumNumberOfConnectedBSCs_2y9Cxd6R9B5xNzs7Z2VO_B/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="MaximumNumberOfConnectedBSCs_2y9Cxd6R9B5xNzs7Z2VO_B" Name="MaximumNumberOfConnectedBSCs" UniqueId="bc24cee7-19b2-4b17-b5fd-d878c27d8f8b" PrimaryDataType="IfcInteger" />
```
No need for change

czapplitec commented on IFC4x3/Properties/m/MaximumNumberOfRRUs_0tAsowuCH8xgtZ_ZgM$4Ug/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="MaximumNumberOfRRUs_0tAsowuCH8xgtZ_ZgM4Ug" Name="MaximumNumberOfRRUs" UniqueId="372b6cba-e0c4-48ee-ade3-fa3a96fc47aa" PrimaryDataType="IfcInteger" />
```
No need for change

czapplitec commented on IFC4x3/Properties/n/NumberOfAntennas_39HzNd1Dv8guqqzDmmmPhJ/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfAntennas_39HzNd1Dv8guqqzDmmmPhJ" Name="NumberOfAntennas" UniqueId="c947d5e7-04de-48ab-8d34-f4dc30c19ad3" PrimaryDataType="IfcInteger" />
```
No need for change

czapplitec commented on IFC4x3/Properties/n/NumberOfAntennas_3pcohitgD84wYzTL$rzTOJ/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfAntennas_3pcohitgD84wYzTLrzTOJ" Name="NumberOfAntennas" UniqueId="f39b2aec-dea3-4813-a8bd-755ff5f5d613" PrimaryDataType="IfcInteger" />
```
Already have replied too much :)

czapplitec committed:
Remove redundancy and improve documentation

czapplitec committed:
Remove redundent Penums and Pconstants

czapplitec committed:
Change property type

jmirtsch committed:
Explicit IfcLabel PrimaryDataType

czapplitec committed:
Harmonize "NumberOfInterfaces"

czapplitec committed:
Add documentation for Psets

In my view, all should be fine. Maybe should consider defining Pset_MobileTelecommunicationsApplianceTypeCommon if we do this for all elements.

czapplitec commented on IFC4x3/PropertyEnumerations/r/PEnum_RemoteUnitConnectionType.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PEnum_RemoteUnitConnectionType" Name="PEnum_RemoteUnitConnectionType" UniqueId="960f2795-0ae4-4243-80ef-119be5b903da">
+	<Constants>
+		<DocPropertyConstant xsi:nil="true" href="CHAIN_3so8ETHqLFfRIPoztV5GAO" />
```
All these redundant constants should have been solved

larswik commented on IFC4x3/Properties/a/AntennaType_2PysPrqKj3aPWcff71DZsA/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="AntennaType_2PysPrqKj3aPWcff71DZsA" Name="AntennaType" UniqueId="99f36675-d14b-4391-9826-a691c1363d8a" />
+
```
PrimaryDataType still missing.

czapplitec commented on IFC4x3/Properties/n/NumberOfInterfaces_3$jLpNUj92SQ9VcAUpjI_k/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfInterfaces_3jLpNUj92SQ9VcAUpjI_k" Name="NumberOfInterfaces" UniqueId="ffb55cd7-7ad2-4271-a25f-98a7b3b52fae" PropertyType="p_tablevalue" />
```
The redundant one is removed

czapplitec commented on IFC4x3/Properties/n/NumberOfInterfaces_3$jLpNUj92SQ9VcAUpjI_k/Documentation.md:
```diff
@@ -0,0 +1 @@
+Indicate the types of interfaces and numbers of each type of them of the device.
```
see above

czapplitec commented on IFC4x3/PropertyConstants/c/CHAIN_3so8ETHqLFfRIPoztV5GAO.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="CHAIN_3so8ETHqLFfRIPoztV5GAO" Name="CHAIN" UniqueId="f6c8839d-4745-4fa5-b499-cbdddf150298" />
```
The duplicated one is removed.

czapplitec committed:
Final check

Status: ready to merge if no further comments.

SergejMuhic reviewed (commented):
Additionally to the comments:
- constants missing documentation
- enumerations missing documentation

czapplitec committed:
Apply suggestions from code review

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

czapplitec commented on IFC4x3/Properties/n/NumberOfEmergencyTransceivers_2RrzYcZZf2sO3dp5bbzVgl/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfEmergencyTransceivers_2RrzYcZZf2sO3dp5bbzVgl" Name="NumberOfEmergencyTransceivers" UniqueId="9bd7d8a6-8e3a-42d9-80e7-cc5965f5faaf" PrimaryDataType="IfcInteger" />
```
No. Will not be modelled in that detail.

czapplitec commented on IFC4x3/Properties/b/BandWidth_3qGbr0Q199cuUUYiO3nY0i/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="BandWidth_3qGbr0Q199cuUUYiO3nY0i" Name="BandWidth" UniqueId="f4425d40-6812-499b-879e-8ac603c6202c" PrimaryDataType="IfcFrequencyMeasure" />
```
Yes, double checked with domain experts, it should be Hz. This is the definition of Band Width in signal processing: https://en.wikipedia.org/wiki/Bandwidth_(signal_processing)

czapplitec commented on IFC4x3/Properties/n/NumberOfInterfaces_3D0wHuI4P3guCzHoS3Y4Zf/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfInterfaces_3D0wHuI4P3guCzHoS3Y4Zf" Name="NumberOfInterfaces" UniqueId="cd03a478-4846-43ab-833d-4727038848e9" PropertyType="p_tablevalue" PrimaryDataType="IfcLabel" SecondaryDataType="IfcInteger" />
```
Same. BBU is just a device. This property is used to same how many interfaces it has.

czapplitec commented on IFC4x3/Properties/t/TransmittedBandwidth_2x5FCzOXX65wSk8PZymLiB/Documentation.md:
```diff
@@ -0,0 +1 @@
+Indicates the transmitted bandwidth of the master unit.
```
Double checked again. It should also be Hz, same with BandWidth.

czapplitec replied:
Resolved: https://github.com/bSI-InfraRoom/IFC-Specification/pull/158/commits/6517df5dcde9580e8fafc2cb2dd05c7a8b4e42ff

czapplitec committed:
Update

czapplitec commented on IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_MobileTeleCommunicationsApplianceTypeRemoteRadioUnit/Documentation.md:
```diff
@@ -0,0 +1 @@
+Properties common to the definition of all occurrences of  _IfcMobileTelecommunicationsAppliance_ and types of _IfcMobileTelecommunicationsApplianceType_ with the predefined type set to REMOTERADIOUNIT.
```
OK, I got the point. Resolved: https://github.com/bSI-InfraRoom/IFC-Specification/pull/158/commits/6517df5dcde9580e8fafc2cb2dd05c7a8b4e42ff

SergejMuhic committed:
Merge branch 'develop' into RailProperties_IfcMobileTelecommunicationsAppliance
